lib/full-site-editing.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Full Site Editing Utils
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Returns whether the current theme is an FSE theme or not.
+ *
+ * @return boolean Whether the current theme is an FSE theme or not.
+ */
+function gutenberg_is_fse_theme() {
+	return is_readable( locate_template( 'block-templates/index.html' ) );
```

Comment:
`locate_template` is doing some more stuff like calling `load_template` and requiring the found file. Do we really want all of that here, as opposed to just checking for the presence of the file?

Reply:
Good question, this is inspired by the work done for theme.json here https://github.com/WordPress/gutenberg/pull/20047/files#diff-0b8e98dead5542e95159bd3c482158111d3f16637e7eef9af13373358fa42dedR14 by @nosolosw 

I didn't give it many thoughts. @nosolosw any particular reason for this there too?

Reply:
I think it's fine given that the template won't be loaded unless you do `locate_template( template, true );`, which is not the case here ([see](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/template.php#L657)). However, if there's a prefered way to check for the file, that'd be good as well.

Reply:
I'm thinking that something as simple as

```php
file_exists( STYLESHEETPATH . '/block-templates/index.html' );
```

should work. I don't think we need `theme-compat` and `TEMPLATEPATH` fallbacks for new themes.